### ISSUE TITLE
No callBackURL

### ISSUE THREAD

**Issue by louisefindlay23** [baahrens/goodreads-api-node#22]
I'm having some issues using OAuth authentication and would appreciate any help you could provide. I keep getting the error `initOAuth(): Warning: You have passed no callbackURL.` I'm using the same URL I've defined when I created my API keys.

```
var callbackURL = "http://127.0.0.1/goodreads";
const gr = goodreads(myCredentials, callbackURL);
```
```
app.get("/authenticate", function (req, res) {
    gr.getRequestToken()
        .then(url => {
            console.log(url);
            res.redirect(url);
        }).catch(function () {
            console.log("Promise Rejected");
        });
});
```

**Comment by louisefindlay23** [comment]
I can successfully authenticate using passport-goodreads and get an oauth_token but how do I pass into the gr.getAccessToken()?

**Comment by baahrens** [comment]
Hi!
Not sure what is happening in the first case. Are you sure, you're not calling `initOAuth` somewhere? That shouldn't happen, if it's really just those two lines you provided.

Also you don't need to pass the `oauth_token` to `getAccessToken`, that is handled internally. 

Hope that helps!

**Comment by louisefindlay23** [comment]
Hi. 

I don't think initOAuth is anywhere. I made a minimal example here: https://github.com/louisefindlay23/library-trackr/blob/api-only/app.js which triggers the issue.

Thanks for your advice.

**Comment by baahrens** [comment]
Thanks for the example code!

First off: It's never a good idea to publicly share your API key/secret pair.

The example seems to work if you remove the leading space in `secret` and set the callbackURL manually:
```js
var callbackURL = "http://127.0.0.1/goodreads";
const gr = goodreads(myCredentials);
gr.initOAuth(callbackURL)
```

I'll investigate this further, but you should be able to work with this for now. 🔮

**Comment by louisefindlay23** [comment]
Thanks a lot, your code snippet fixes the error and I've moved my keys to a .env file. Beforehand, the repo was private.

~~However, once the getRequestToken is complete, it redirects to a https version of the callback url instead of http. Not sure why since I specified in the code and in the Goodreads API key, http.~~ Fixed by installing https-localhost and creating a self-signed certificate.
